### ISSUE TITLE
Install and configure MVP plugin from available open source plugin

### ISSUE THREAD

**Issue by laineymajor** [department-of-veterans-affairs/va.gov-team#45300]
**Problem Statement**
The Platform Console will be used as a one-stop shop for VFS developers to complete the work for their app(s). To make the Platform Console more useful, more services need to be made available, including products/services that are owned by Release Tools.

How might we increase the number of services offered on Platform Console by leveraging existing open source plugins on backstage [that are specifically related to services/products offered by the Release Tools team]?

**Hypothesis or Bet**
If more services are made available on the Platform Console, it will be more utilized and useful by VFS app teams.
If we find open source plugins that are applicable to our team's services/products, we, and potentially other teams, will add more plugins

**Definition of done**
- [ ] Tech Radar plugin has been installed and appropriately configured

**Comment by laineymajor** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VSP---Release-Tools-FE-5fc9325744944e0015ed1861/issues/department-of-veterans-affairs/va.gov-team/45300?planning-poker) with ZenHub @jhonnyoddball @ndsprinkle

**Comment by ndsprinkle** [comment]
### Tech Radar v1 Overview

#### Quadrants

Languages - The most straightforward quadrants, represents all programming languages
Frameworks - Also fairly straightforward, represents all frameworks, usually the entry will call itself a framework
Infrastructure - I view this as representing Ops related entries, like deployments (Jenkins/GitHub actions)
Process - Probably the vaguest and could use a reworking in v2, but represents anything that's a part of the process of making changes to vets-website, which is mainly just code reviews in GitHub

#### Rings

Rings in order of innermost to outermost:

IN USE - Represents entries that vets-website currently utilizes
IN TESTING - Represents entries that are being tested to see if they are a good fit for vets-website, their goal is to become IN USE
DEPRECATED - Represents entries that were previously IN USE, but have been replaced
UNAPPROVED - Represents entries that were **never** IN USE, but are recognized as not being a good fit for vets-website

#### Entries

Entries are the actual nodes on the tech radar. Each entry is located within one quadrant and one ring.

| Entry | Quadrant | Ring |
|---|---|---|
| JavaScript | Languages | IN USE |
| SCSS | Languages | IN USE |
| React | Frameworks | IN USE |
| Redux | Frameworks | IN USE |
| Webpack | Frameworks | IN USE |
| Cypress | Frameworks | IN USE |
| Yarn | Frameworks | IN USE |
| Code Reviews | Process | IN USE |
| GitHub Actions | Infrastructure | IN USE |
| GitHub Codespaces | Infrastructure | IN TESTING |
| Jenkins | Infrastructure | DEPRECATED |
| Jest | Frameworks | UNAPPROVED |

**Comment by ndsprinkle** [comment]
#### Why the Tech Radar plugin doesn't work in the catalog

The console UI team felt that the Tech Radar plugin best belonged in the catalog scoped to a specific entity. In theory, it makes sense to scope the Tech Radar to a system entity that corresponds to a repository, but in practice this doesn't work well with the plugin.

For example, take the [Sentry plugin](https://github.com/backstage/backstage/tree/master/plugins/sentry) that we reviewed. In the [Getting Started](https://github.com/backstage/backstage/tree/master/plugins/sentry#getting-started) section, the plugin describes two provided elements: `EntitySentryCard` and `EntitySentryContent`. Both of these plugins are intended to be added to the `EntityPage`, which is where we modify the entity page for a given entity in the catalog. These elements are designed specifically for the catalog. The card element is designed for the entity overview page and the content element is designed for a tabbed, full page view within an entity in the catalog.

Conversely, if we look at the [Configuration](https://github.com/backstage/backstage/tree/master/plugins/tech-radar#configuration) for the Tech Radar plugin, the only provided element is `TechRadarPage`. This is not designed for use in the catalog but rather for use on a dedicated page. There are no provided entity elements for the Tech Radar plugin making its use in the catalog not ideal and nearly impossible.

For most people, that would be reason enough to give up on putting the Tech Radar in the catalog, but I pressed on. It is theoretically possible to use that `TechRadarPage` in the catalog, so I shoved it into the entity page in its own tabbed view. The problem is the design, it just looks completely off. The main issue is that the plugin renders a header:

 ![image.png](https://images.zenhubusercontent.com/5d5ebd091325bafe068c37dc/8b5d0c4b-e1db-458c-8ef1-b157e7075bb5)

So you get double header thing that clearly isn't correct. Again, the header is present because it is designed to live on its own dedicate page somewhere, not the catalog.

Lastly, the Tech Radar plugin doesn't provide a way to conditionally show data, which is very important for the catalog. The Sentry component supports a `isSentryAvailable` function that can be used to conditionally show the Sentry plugin card or content depending on if there is a Sentry configuration for a given entity. Because the Tech Radar component does not support any conditional logic like this, there's no way to control which Tech Radar is shown for a given entity or to hide the Tech Radar tab/content if there is no Tech Radar data to show.

In conclusion, the Tech Radar plugin we have implemented is not designed for use in the catalog. If we wanted to implement an open source plugin in the catalog, we would need to select something else like the Sentry plugin. Because we have already finished implementation of the Tech Radar plugin, I think it's best to continue down this path, especially considering that the original goal was to simply test the waters of open source Backstage plugins and get something working. In the future, if we want to add a plugin to the catalog, we can select a plugin that is designed for that.

**Comment by ndsprinkle** [comment]
@laineymajor At this point, the last step before making a PR is finalizing the location of the Tech Radar as well as how to access the page. Currently, I simply added a link in the side nav that takes you directly to /tech-radars/vets-website. The long term plan was that this side nav could expand to a sub-nav that contained links to all the different Tech Radars. Was there someone that you had recommended I talk to about adding the link to the side nav?

Alternatively, we could drop the side nav idea altogether and add a link within the vets-website catalog entity. There's a small section of links that lives on the overview page for a given entity and a link to this Tech Radar page could potentially just live there. That would save us the trouble of messing with the side nav. It's extremely easy to update the links, you just make changes here: https://github.com/department-of-veterans-affairs/vets-website/blob/main/catalog-info.yaml#L12

I think the only issue I could see with this is that all the other links are external and this link to the Tech Radar page would be internal? I don't know if that matters at all, but I suppose it would be worth researching the Backstage docs to see if these links are intended to be external only, but I highly doubt it matters. Beyond that it obviously means the Tech Radar and future Tech Radars are much more hidden and take longer to access, but again this may be a non-issue since it seemed like the immediate feedback was that the side nav was too front-and-center.
